### PR TITLE
Some changes to Sampling Op

### PR DESCRIPTION
**Pull request by wangyems**
### Description

1. add an optional input to pass in seed
2. two UTs. one for top_p=0.5, another for top_p=0.01(create greedy search result, in convert_generation.py)
3. fix a bug in cpu kernel

### CODE REVIEW

File: onnxruntime/test/contrib_ops/sampling_test.cc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  std::vector<Ort::Value> ort_inputs;
+  ort_inputs.push_back(std::move(input_ids_tensor));
+  ort_inputs.push_back(std::move(max_length_tensor));
+  ort_inputs.push_back(std::move(min_length_tensor));
+  ort_inputs.push_back(std::move(repetition_penalty_tensor));
+  const char* input_names[] = {"input_ids", "max_length", "min_length", "repetition_penalty"};
+  const char* const output_names[] = {"sequences"};
+
+  constexpr int min_cuda_architecture = 530;
+  if (HasCudaEnvironment(min_cuda_architecture)) {
```
Comment (yufenglee) [inline onnxruntime/test/contrib_ops/sampling_test.cc:57]: Add verification on cpu

File: onnxruntime/python/tools/transformers/convert_generation.py [hunk 2]
```diff
@@ -1709,6 +1722,10 @@
         )
         graph_inputs.append(presence_mask)
 
+    if is_sampling and args.seed:
+        seed = onnx.helper.make_tensor_value_info("seed", TensorProto.INT32, [1])
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/convert_generation.py:1726]: >seed [](http://example.com/codeflow?start=51&length=4)

How about to make seed configurable? default set to -1. if >0, use the seed passed in by args

File: onnxruntime/test/contrib_ops/sampling_test.cc (hunk 1 above)
Comment (wangyems) [reply onnxruntime/test/contrib_ops/sampling_test.cc:57]: found some issue with cpu. investigating..

File: onnxruntime/python/tools/transformers/convert_generation.py (hunk 2 above)
Comment (wangyems) [reply onnxruntime/python/tools/transformers/convert_generation.py:1726]: currently we only test sampling with default seed and small top_p. so this flag limits to help create the model with that input

File: onnxruntime/test/contrib_ops/sampling_test.cc [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+#include "core/common/gsl.h"
+#include "core/session/onnxruntime_cxx_api.h"
+#include "test/common/cuda_op_test_utils.h"
+
+extern std::unique_ptr<Ort::Env> ort_env;
+
+namespace onnxruntime {
+namespace test {
+
+#if defined(__linux__) && !defined(__ANDROID__)
```
Comment (wangyems) [inline onnxruntime/test/contrib_ops/sampling_test.cc:16]: Limit the test to Linux machines since random number is generated machine-specific

**Review by yufenglee — APPROVED** [review]
:shipit:

**Pull request merged**